mailpoet/assets/js/src/segments/dynamic/dynamic.tsx

```diff
@@ -8,14 +8,36 @@ import { Editor } from 'segments/dynamic/editor';
 import { DynamicSegmentList } from 'segments/dynamic/list';
 import { SegmentTemplates } from 'segments/dynamic/templates';
 import * as ROUTES from 'segments/routes';
-import { createStore } from 'segments/dynamic/store';
+import { createStore, storeName } from 'segments/dynamic/store';
+import { useEffect, useRef } from 'react';
+import { useDispatch } from '@wordpress/data';
 
 const container = document.getElementById('dynamic_segments_container');
 
+function HistoryListener() {
+  const { setPreviousPage } = useDispatch(storeName);
+  const history = useHistory();
+
+  const previousPageRef = useRef(history.location.pathname);
+
+  useEffect(
+    () =>
+      history.listen((location) => {
+        setPreviousPage(previousPageRef.current);
+
+        previousPageRef.current = location.pathname;
+      }),
+    [history, setPreviousPage],
+  );
+
+  return null;
+}
```

Comment:
If I understand this correctly, we look for _any_ previous page. In that case, could we just use `history.back()` or look at the browser history directly (`document.referrer`), when needed?

On the other hand, thinking further, this listener will save only pages withing this React app, which is probably what we want and that's why we need this hook, right?

Reply:
I opted to use this listener to save only pages within this React app and also because this was the way that I found to have a default value and point the users to the listing page when they opened the segment edit page directly.